### PR TITLE
Add first draft of TOTP-mismatch.

### PR DESCRIPTION
**Pull request by Siproqu**
Any thoughts on the content?

Annotations are´(formatted like this)´.

### CODE REVIEW

**Comment by tlaurion** [comment]
A bit long, love added references.
I think the administrator should be removed. User should know the source of the build and know its authenticity and flash it himself since upgrading requires resealing,and setting a new boot default.
This is really where one can see measurements differences, when a user types his disk unlock key passphrase correctly and that it doesn't unlock the encrypted drive. This is where measurements of parts are shown and if a user has a screenshot of those measurements, can actually see which part changed. The TPM will not release the secret to unlock the disk if the luks header changed, if rom measurements + public key are not the same, and if paraphrase is not good to unlock TPM NVRAM space. Also the TPM will not release the secret, and will change TPM PCRs if it went to recovery on the boot path.

 Plus, booting the default boot option verifies detached /boot signed digest.

Unfortunately,  if the TOTP went wrong, measurements changed and the laptop should not be trusted. Grab your backups (you have backups right?), reflash rom and reinstall and wonder what happened between last remote attestation and now, wonder on your real threat model and reinstall while restoring backups in emergency restore mode. 


Solution would be to externally reflash. This is where x230-maximized will come into play since it permits to reflash both spi chips in one run, externally.

Will merge those soon. Buids are available on circleci for each commit. 

On November 27, 2020 11:53:17 PM UTC, Siproqu <notifications@github.com> wrote:
>Any thoughts on the content?
>
>
>
>Annotations are´(formatted like this)´.
>
>You can view, comment on, or merge this pull request online at:
>
>
>
>  https://github.com/osresearch/heads-wiki/pull/53
>
>
>
>-- Commit Summary --
>
>
>
>  * Add first draft of TOTP-mismatch.
>
>
>
>-- File Changes --
>
>
>
>    A Help/TOTP-mismatch.md (85)
>
>
>
>-- Patch Links --
>
>
>
>https://github.com/osresearch/heads-wiki/pull/53.patch
>
>https://github.com/osresearch/heads-wiki/pull/53.diff
>
>
>
>-- >
>You are receiving this because you are subscribed to this thread.
>
>Reply to this email directly or view it on GitHub:
>
>https://github.com/osresearch/heads-wiki/pull/53
>

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Siproqu** [comment]
@tlaurion Do think it has still to many unnecessary details? I am not really sure who the target audience of this wiki is. Professionals, enthusiasts, technical people who come from a different field?

Text to the third point will follow.

File: Help/TOTP-mismatch.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+**3. Your firmware has been manipulated.**
+**Please note: This possibility is the most unprobable. Make sure that you have followed thoroughly the steps in solution 1!
+
+
+
+More details about the issues cause.
+---
+
+´(Add tlaurions draft here? It is really well written.)´
```
Comment (tlaurion) [inline Help/TOTP-mismatch.md:66]: @Siproqu take what you want!

**Review by tlaurion — CHANGES_REQUESTED** [review]

**Comment by Siproqu** [comment]
Will open it again when ready.

**Comment by githubuseravailable** [comment]
@tlaurion @Siproqu 

i have replaced the RTC Battery, and tried the 1st solution in [TOTP Mismatch](https://github.com/osresearch/heads-wiki/pull/78/commits/6953c96d7b5fc43a0a1140b3e8a46162e3c3f63a)

date -s hh:mm:ss && hwclock -w

it has set the current HW clock to https://time.is/GMT
but it doesn't set the date, only the HW clock, 
and the TOTP still mismatch,

is there any way to set the date too ?

**Pull request closed without merging**